src/main/java/org/folio/circulation/domain/Item.java

```diff
@@ -159,33 +152,13 @@ public String getInstanceId() {
     return getProperty(holdingRepresentation, "instanceId");
   }
 
-  public String getCallNumber() {
-    return getEffectiveCallNumberProperty(CALL_NUMBER_ID);
-  }
-
-  public String getCallNumberPrefix() {
-    return getEffectiveCallNumberProperty(CALL_NUMBER_PREFIX_ID);
-  }
-
-  public String getCallNumberSuffix() {
-    return getEffectiveCallNumberProperty(CALL_NUMBER_SUFFIX_ID);
-  }
-
-  private String getEffectiveCallNumberProperty(String propertyName) {
-    return hasItemRepresentationCallNumber()
-      ? getProperty(itemRepresentation, mapToItemCallNumberPropertyName(propertyName))
-      : getProperty(holdingRepresentation, propertyName);
-  }
-
-  private boolean hasItemRepresentationCallNumber() {
-    return StringUtils.isNotBlank(getProperty(itemRepresentation, ITEM_CALL_NUMBER_ID));
-  }
+  public EffectiveCallNumberComponents getEffectiveCallNumberComponents() {
+    if (!itemRepresentation.containsKey(EFFECTIVE_CALL_NUMBER_COMPONENTS)) {
+      return null;
+    }
 
-  private String mapToItemCallNumberPropertyName(String holdingsPropertyName) {
-    return Stream.of(ITEM_CALL_NUMBER_ID, ITEM_CALL_NUMBER_PREFIX_ID, ITEM_CALL_NUMBER_SUFFIX_ID)
-      .filter(val -> StringUtils.containsIgnoreCase(val, holdingsPropertyName))
-      .findFirst()
-      .orElse(StringUtils.EMPTY);
+    return itemRepresentation.getJsonObject(EFFECTIVE_CALL_NUMBER_COMPONENTS)
+      .mapTo(EffectiveCallNumberComponents.class);
```

Comment:
I'd like for mod-circulation to move away from accessing JSON representations within the domain. (I'm aware this is a big task, which is why my current focus is on not making it any bigger).

Could this be mapped to a class during instantiation of the `item` class, meaning that the rest of the code is unaware of the JSON representation?